### PR TITLE
[Database] Define DatabasePersistentStorageTests

### PR DESCRIPTION
**Pull request by Endritz**
Working on #60

### CODE REVIEW

File: docs/uml-class-diagrams/database/test/DatabasePersistentStorageImplementationTests/DatabasePersistentStorageImplementationTests.puml [hunk 1]
```diff
@@ -2,7 +2,14 @@
...
     class DatabasePersistentStorageImplementationTests {
-
+        + void rfidDataInIsRfidDataOut()
+        + void offsetMatchDataPoint()
+        + void locationDataPointInIsDataPointOut()
+        + void offsetMatchTrackData()
+        + void trackDataInIsTrackDataOut()
+        + void readModeInIsModeOut()
+        + void writeModeInIsModeOut()
+        + void currentSatMatchNextSat()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/database/test/DatabasePersistentStorageImplementationTests/DatabasePersistentStorageImplementationTests.puml:12]: Just a few more test cases regarding edge cases. Odds are we shouldn't hit the first two, but I think an out-of-bounds exception could be likely if we don't make a deliberate effort to handle it.
```suggestion
        + void currentSatMatchNextSat()
        + void cannotStoreNullLocationDataPoint()
        + void cannotStoreNullTrackData()
        + void locationDataPointOffsetOutOfBoundsReturnsNull()
        + void trackDataOffsetOutOfBoundsReturnsNull()
        + void nonexistentSatelliteNameReturnsEmptyNext()
        + void lastSatelliteNameInListReturnsEmptyNext()
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Looks good! I just recommend adding a few more test cases to cover some of the edge cases and make sure that we don't break things when it comes to indexing or serializing and deserializing. Feel free to add more test cases as you get building as well if you think of other conditions that might arise!

**Review by kylepoterek — APPROVED** [review]

**Review by andrewdimmer — APPROVED** [review]
Looks good to me! All set to merge!
